extensions/gadgets/ScoreBar/protractor.js

```diff
@@ -0,0 +1,43 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end testing utilities for the ScoreBar gadget.
+ */
+
+var objects = require('../../objects/protractor.js');
+
+var customizeGadget = function(elem, title, maxVal, targetParam) {
+  elem.element(by.name('form.gadgetSchemaForm')
+    ).all(by.tagName('schema-based-unicode-editor')).then(function(items) {
+      objects.UnicodeStringEditor(items[0]).setValue(title);
+      objects.UnicodeStringEditor(items[1]).setValue(targetParam);
+    });
+
+  objects.IntEditor(
+    elem.element(by.tagName('schema-based-int-editor'))
+  ).setValue(maxVal);
+};
+
+var expectGadgetPreviewDetailsToMatch = function(elem, title) {
+  expect(elem.element(by.css('.protractor-test-scorebar-title')).getText()).toBe(title);
```

Comment:
Ah, I see Math/protractor.js tests for the value of 'raw_latex-with-value' attribute. I thought we'd want to test the value of the variable after it's re-surfaced by the gadget. For example, using:
  expect(elem.element(by.css('.protractor-test-scorebar-progressbar'))
    .getAttribute('max')).toBe(String(maxVal);

Where maxVal is passed in from editorAndPlayer.js with a dynamic value we expect to see. Attached a screenshot of the gadget preview in the editor view.

![getattribute max](https://cloud.githubusercontent.com/assets/1622682/9872049/b8bddde2-5bd1-11e5-95b2-e57af2002f43.png)

As you suggest, I'd like to test user-visible aspects such as the actual value returned by getScoreValue(). getAttribute currently returns a string with the function name instead of the value returned by that function. If this isn't possible, I can add a couple assertions that attribute_name-with-value attributes show up properly, but those are a few steps away from user-visible effects.